### PR TITLE
Removes an obsolete fallback for Python 2.6 in wait_for.

### PR DESCRIPTION
**Pull request by atombrella**
##### SUMMARY

Removing some dead code due to Python 2.6 being the minimum.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
core

### CODE REVIEW

**Comment by atombrella** [comment]
@samdoran Done. My branch is open if you want to re-phrase it.

File: changelogs/fragments/63988-removes-python_compat_fallback.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - code - removes some Python compatibility code for dealing with socket timeouts in wait_for .
```
Comment (samdoran) [inline changelogs/fragments/63988-removes-python_compat_fallback.yml]:
```suggestion
  - code - removes some Python compatibility code for dealing with socket timeouts in ``wait_for``
```

File: lib/ansible/modules/utilities/logic/wait_for.py [hunk 2]
```diff
@@ -447,14 +446,7 @@ def _create_connection(host, port, connect_timeout):
...
     """
-    if sys.version_info < (2, 6):
-        (family, _) = (_convert_host_to_ip(host))[0]
-        connect_socket = socket.socket(family, socket.SOCK_STREAM)
-        connect_socket.settimeout(connect_timeout)
-        connect_socket.connect((host, port))
-    else:
-        connect_socket = socket.create_connection((host, port), connect_timeout)
-    return connect_socket
+    return socket.create_connection((host, port), connect_timeout)
```
Comment (samdoran) [inline lib/ansible/modules/utilities/logic/wait_for.py]: This function can be removed altogether. Change calls to `_create_connection()` to call `socket.create_connection()` directly.

**Review by samdoran — CHANGES_REQUESTED** [review]

**Comment by atombrella** [comment]
@samdoran Thanks. Done :)

**Review by samdoran — APPROVED** [review]

**Pull request merged**